### PR TITLE
[DEV-1660] Add tab component

### PR DESCRIPTION
**Pull request by tommaso1**
#### List of Changes

Add a TabComponent to handle tabs in a generic way

#### Motivation and Context

It will be used for homepage product cards

#### How Has This Been Tested?

Via storybook `Atoms/TabComponent
`
#### Screenshots (if appropriate):
Screen Shot 2024-06-04 at 12 45 28 PM

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/storybook-app/stories/atoms/TabComponent.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          />
+        },
+        {
+            title: 'Title of the second page',
+            content: <div>
+                <h1>Second Page</h1>
+                <p>Some content here</p>
+            </div>
+        }
+    ]
```
Comment (MikeAtUqido) [inline apps/storybook-app/stories/atoms/TabComponent.stories.tsx:30]:
```suggestion
     items: [
      {
        title: "Title",
        content: (
          <IconWrapper
            icon="https://cdn.dev.developer.pagopa.it/app_Io_d9bffd556b.svg"
            isSvg={true}
          />
        ),
      },
      {
        title: "Title of the second page",
        content: (
          <div>
            <h1>Second Page</h1>
            <p>Some content here</p>
          </div>
        ),
      },
    ],
```

File: apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import React, { FC, ReactNode } from 'react';
+import { Box, Tab, Tabs, Typography } from '@mui/material';
+
+export type TabPanelProps = {
+  readonly children?: ReactNode;
+  readonly index: number;
+  readonly value: number;
+};
+
+const TabPanel: FC<TabPanelProps> = ({ children, value, index, ...other }) => {
```
Comment (MikeAtUqido) [inline apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:11]: I think `other` doesn't contain anything, am I wrong?

File: apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  readonly title: string;
+  readonly content: ReactNode;
+};
+
+export type TabsProps = {
+  readonly items: ReadonlyArray<TabItem>;
+};
+
+export const TabComponent: FC<TabsProps> = ({ items }: TabsProps) => {
+  const [value, setValue] = React.useState(0);
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:35]: I think you should use a better name like "currentTab" or "currentTabIndex"

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx (hunk 2 above)
Comment (tommaso1) [reply apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:11]: Yes, it was there to optionally pass some style or classes, but we can add it later if we need it

**Review by MikeAtUqido — APPROVED** [review]

**Review by marcobottaro — APPROVED** [review]

File: apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
+'use client';
+import React, { FC, ReactNode } from 'react';
+import { Box, Tab, Tabs, Typography } from '@mui/material';
+
+export type TabPanelProps = {
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:5]: Why do you export this type?
```suggestion
type TabPanelProps = {
```

File: apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      hidden={value !== index}
+      id={`tabpanel-${index}`}
+      aria-labelledby={`tab-${index}`}
+    >
+      {value === index && <Box p={3}>{children}</Box>}
+    </div>
+  );
+};
+
+export type TabItem = {
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:24]: Why do you export this type?
```suggestion
type TabItem = {
```

File: apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
+'use client';
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:1]: Is `use client` necessary?

File: apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import React, { FC, ReactNode } from 'react';
+import { Box, Tab, Tabs, Typography } from '@mui/material';
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:3]:
```suggestion
import { Box, Tab, Tabs } from '@mui/material';
```

File: apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+export type TabsProps = {
+  readonly items: ReadonlyArray<TabItem>;
+};
+
+export const TabComponent: FC<TabsProps> = ({ items }: TabsProps) => {
+  const [currentTab, setCurrentTab] = React.useState(0);
+
+  return (
+    <div>
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:37]: Do we need this `<div>`? Maybe a fragment is enough
```suggestion
    <>
```

File: apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        {items.map((item, index) => (
+          <Tab key={index} label={item.title} />
+        ))}
+      </Tabs>
+      {items.map((item, index) => (
+        <TabPanel key={index} value={currentTab} index={index}>
+          {item.content}
+        </TabPanel>
+      ))}
+    </div>
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:53]:
```suggestion
    </>
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx (hunk 6 above)
Comment (tommaso1) [reply apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:1]: Yes, because of MUI

**Review by marcobottaro — APPROVED** [review]

**Review by marcobottaro — APPROVED** [review]

**Pull request merged**